### DISCUSSION TITLE
Can we put the Invoice Payment Methods response data inside the Get Invoice(s) response? Why a separate call?

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2678] [Technical Support]
I noticed that to get the payment information of an invoice, you need to use the "Get Payment Methods" call.

This "resource" only has 1 call, and it's a GET call, so why not simply put the response inside the "Get Invoice" or "Get Invoices" calls in a nested fashion?

If there were POST or DELETE methods on the "Payment Methods" resource, it would make sense to use a separate URL as you would be doing more actions on it, but since you can only GET this data, putting it directly in the invoice response would make a lot of sense.

Unless there are some future changes I am missing that is...

Why am I asking this? While working on the Zapier integration, I need to be mindful of the number of calls I make. I cannot make extra calls for each invoice as time is limited. If the data would be presented in 1 go, that would help performance and usability... unless this call is costly on its own, then yes, I understand to keep it separate.

To make this into Zapier, I now need to make a separate Action that just loads this data and loop it. It isn't very handy for me or the Zapier user.

I know, breaking API specs is dangerous, so we could add a flag that tells the API to include the payment methods nested? Or maybe keep for Greenfield v2 ?

Curious about your thoughts @Kukks and @NicolasDorier

**Comment by bolatovumar** [comment]
Sounds reasonble-ish (unless @Kukks or @NicolasDorier know why it wouldn't be). 

> I know, breaking API specs is dangerous, so we could add a flag that tells the API to include the payment methods nested? 

Aren't you just adding stuff here so it wouldn't break anything? Nothing is removed, right?

**Reply by woutersamaey** [reply]
Correct. Only adding. So it is safe. Just unexpected I guess.
